Adds scheduled, and manual test runner, for the Dapr Java + Python SDK test validation.

JoshVanL opened the pull request:
Part of #5797.

Adds GitHub action workflow for running SDK tests against Dapr HEAD. These are run on an infrequent schedule, as well as on pull requests when someone comments `/test-sdk-all`.

Tests  the Java SDK, which can also be triggered with `/test-sdk-java`.

Tests  the Python SDK, which can also be triggered with `/test-sdk-python`.

Not fully tested e2e just because of how GH actions work, but hopefully should be close to working.

/cc @artursouza

JoshVanL commented:
/test-sdk-java

artursouza commented on .github/workflows/dapr-test-sdk.yml:
```diff
@@ -0,0 +1,129 @@
+  GOOS: linux
+  GOARCH: amd64
+  GOPROXY: https://proxy.golang.org
+  DAPR_CLI_VER: 1.10.0
```
Can you install the latest Dapr CLI?

artursouza commented on .github/workflows/dapr-test-sdk.yml:
```diff
@@ -0,0 +1,129 @@
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: true
+      matrix:
```
Please, pick only one variation. This will save in the total number of runs in parallel.

artursouza commented on .github/workflows/dapr-test-sdk.yml:
```diff
@@ -0,0 +1,129 @@
+      run: cd java-sdk && mvn compile -B -q
+    - name: Unit tests
+      run: cd java-sdk && mvn -B test -q
+    - name: Codecov
```
We can remove code coverage.

artursouza reviewed (commented):
Can you push this to your fork's master and paste the link of the run over there?

berndverst commented:
If you want I can push a commit to this to add Python validation also.

berndverst commented:
I took the liberty to push a Python workflow to your PR and fix an issue in your PR.

I reduced the Java runs - but I don't know which versions you want, please update accordingly!

Also I noticed that the `/test-sdk-python` for example still triggers the java run too, so that logic seems incorrect there.

JoshVanL commented:
Thanks @berndverst. Personally I prefer to have only review comments on my PRs, or a PR to my branch, rather than pushing straight to it. Do make sure to update the PR description too! it's really no worries though, thank you for the work :slightly_smiling_face:

JoshVanL commented:
Thanks both! I have addressed the comments and fixed the issues. Jobs are now properly run using the commands as expected.

You can see these jobs being triggered in my personal fork (note, I believe these failures are true positives):
- https://github.com/JoshVanL/dapr/actions/runs/4882054149
- https://github.com/JoshVanL/dapr/actions/runs/4881969948
- https://github.com/JoshVanL/dapr/actions/runs/4881880280

berndverst commented:
> Thanks @berndverst. Personally I prefer to have only review comments on my PRs, or a PR to my branch, rather than pushing straight to it. Do make sure to update the PR description too! it's really no worries though, thank you for the work 🙂

Absolutely, and I wouldn't usually do that, so definitely feel free to remove my commits.

Given all the issues we have seen in gRPC based actor SDKs and HTTP based actor SDKs - it seemed critical to also enable a HTTP based SDK validation immediately to help with our release testing. In this case I figured it is better to ask for forgiveness. So my apologies!
Having multiple PRs open unnecessarily for essentially duplicate things just causes a significant delay - reviewers don't get to them, and as a PR author you of course need to deal with the conflicts. And then there is the GitHub runner resource shortage too...

Rather than leaving a review comment saying "Also add the Python SDK here" and pointing you to the steps, was trying to simplify that for you. At this stage it is best to reduce the number of PRs.

Won't update anyone's PR in runtime again! Sorry :)

mukundansundar commented on .github/workflows/dapr-test-sdk.yml:
```diff
@@ -0,0 +1,173 @@
+    name: "Java SDK verification tests"
+    runs-on: ubuntu-latest
+    env:
+      JDK_VER: 16
```
```suggestion
      JDK_VER: 11
```
please use 11 ... 16 is not LTS

mukundansundar commented on .github/workflows/dapr-test-sdk.yml:
```diff
@@ -0,0 +1,173 @@
+        run: cd java-sdk && mvn clean -B
+      - name: Build sdk
+        run: cd java-sdk && mvn compile -B -q
+      - name: Unit tests
+        run: cd java-sdk && mvn -B test -q
```
this step can be removed.

mukundansundar reviewed (changes requested):
Please add `/test-sdk-all` and the other ones to the https://github.com/dapr/dapr/blob/master/.github/scripts/dapr_bot.js

artursouza commented:
Why does the Python SDK fail (1) and why does it show a green check when it fails (2)?
